src/DI/ContainerBuilder.php

```diff
@@ -101,16 +107,49 @@ public function getDefinitions()
 
 
 	/**
-	 * Does the service definition exist?
+	 * Does the service definition or alias exist?
 	 * @param  string
 	 * @return bool
 	 */
 	public function hasDefinition($name)
 	{
+		$name = isset($this->aliases[$name]) ? $this->aliases[$name] : $name;
 		return isset($this->definitions[$name]);
 	}
 
 
+	/**
+	 * @param string
+	 * @param string
+	 */
+	public function addAlias($alias, $service)
+	{
+		if (!is_string($alias) || !$alias) { // builder is not ready for falsy names such as '0'
+			throw new Nette\InvalidArgumentException(sprintf('Alias name must be a non-empty string, %s given.', gettype($alias)));
+
+		} elseif (isset($this->aliases[$alias])) {
+			throw new Nette\InvalidStateException("Alias '$alias' has already been added.");
+
+		} elseif (isset($this->definitions[$alias])) {
+			throw new Nette\InvalidStateException("Service '$alias' has already been added.");
+
+		} elseif (!isset($this->definitions[$service])) {
```

Comment:
This check is unnecessary, because you can add definition, add alias, remove definition.